doc/mob_db.txt

```diff
@@ -82,7 +82,7 @@ mob_db: (
     HeadTopId: top headgear id       (int, defaults to 0)
     HeadMidId: middle headgear id    (int, defaults to 0)
     HeadLowId: lower headgear id     (int, defaults to 0)
-    HairStyleId: hair style id       (int, defaults to 0)
+    HairStyleId: hair style id       (int, defaults to 1)
```

Comment:
Maybe add documentation somewhere, that if viewData is defined, then hairstyle will be default to 1

Reply:
In my opinion this isn't necessary. It should be obvious that the fields inside a block will only be assigned if that block is defined. If I'd start documenting this explicitly we'd have to do it for more fields, too.
For example:
 * item_db->Trade->override
 * item_db->Nouse->override
 * job_db->BaseASPD->all fields except *Shield*
 * mob_db->ViewData->SpriteId
 * mob_db->ViewData->Gender
 * refine_db->Rates->Level->all fields except *Bonus*

And why only if it's a field inside a block? Actually we'd have to document that behaviour for every field which doesn't default to 0, then. I don't think that's useful...

Reply:
actually not,
Those override of item_db defaults to 100, even if block is not defined. (for item, itemdb_validate_entry).
For mob_db->ViewData->SpriteID, it is defaulted to mob_read_db_sub just after "Id" is read.
mob_db->ViewData->Gender defaults to SEX_FEMALE (Value: 0)

For RefineDB, even before reading all the fields, default value is set.
```
for (int i = 0; i < REFINE_CHANCE_TYPE_MAX; i++)
			for (int j = 0; j < MAX_REFINE; j++)
				chance[i][j] = 100; // default value for all rates.
```

Reply:
I don't want to argue, but do you really think that a user wants to know that `HairStyleId` won't default to 1, if he doesn't even use the `ViewData` block?
Well, to don't push this topic too far, I add a note in _doc/mob_db.txt_.